### DISCUSSION TITLE
Стоит ли добавлять onClick проп в TabbarItem?

### DISCUSSION THREAD

**Discussion by scffs** [VKCOM/VKUI#5280] [Q&A] [answered]
TS2322: Type '{ selected: boolean; onClick: () => any; "aria-label": string; }' is not assignable to type 'TabbarItemProps'.   Property 'onClick' does not exist on type 'TabbarItemProps'.
```jsx
<TabbarItem selected={text === 'one'} onClick={() => setText('one')} aria-label='Новости'>
  <Icon28NewsfeedOutline />
</TabbarItem>
```

Стоит ли добавлять `onClick: () => void` проп или как решать эту проблему?

В том же компоненте, кстати, есть ссылка на класс, которого нет в стилях

![image](https://github.com/VKCOM/VKUI/assets/67464545/1990b742-3ca8-4a97-8428-02ca7ec5057d)

![image](https://github.com/VKCOM/VKUI/assets/67464545/08ecd3ef-afdd-48ef-91db-8f4fde0a9f7f)

**Comment by inomdzhon** [comment]
Какая у тебя версия VKUI?

> В том же компоненте, кстати, есть ссылка на класс, которого нет в стилях

Да, можно удалить. Будем рады PR ;)

---

PS: ко всем запросам стоит прикреплять версию VKUI

**Reply by scffs** [reply]
Версия последняя
Прикольчик в том, что я открыл проект, чтобы проверить версию и сверить ее с  latest, но забыл ее дописать в сюда, сорянчик)))

UPD
`"@vkontakte/vkui": "^5.5.2",`

**Reply by inomdzhon** [reply, marked as answer]
Аааа ой, ты `src/components/TabbarItem/Readme.md` что ли смотришь? Опять же, это твой IDE пытается раздуплить что там по типам.

Никогда не доверяй IDE))) Если ты работаешь с VKUI, то в корне проекта есть команда `yarn run lint:types` – стоит ориентироваться только на него.

PS1: У нас нет тайп чека в файлах маркдаун по типу `Readme.md`, это же совсем другой формат.
PS2: А сам компонент наследует параметры от `React.AllHTMLAttributes<HTMLElement>` и там есть `onClick` в том числе.

**Reply by scffs** [reply]
Нееет, это не Readme.md, это у меня в проекте такой код. (который взят из ридми, да)

**Reply by scffs** [reply]
![image](https://github.com/VKCOM/VKUI/assets/67464545/6b32ece9-5baa-4f32-ba5e-4ccc93bc1a06)

Ещё на это ругается :(

**Reply by inomdzhon** [reply]
А какой версии TS у тебя?

**Reply by scffs** [reply]
5.1.3

**Reply by inomdzhon** [reply]
Версия `WebStorm` >= 2023?

**Reply by scffs** [reply]
ой сорри
вот версия 
![image](https://github.com/VKCOM/VKUI/assets/67464545/eabca79f-15a4-44e6-9e2c-2496f1caeb96)

**Reply by inomdzhon** [reply]
Тебе надо обновиться до `WebStorm` >= 2023

Typescript 5 **не** поддерживается на версиях младше

**Reply by scffs** [reply]
чекни пж выше коммент

**Reply by inomdzhon** [reply]
Вижу, тогда странно

Если в терминале вызывать `tsc`, то такая же проблема?

**Reply by scffs** [reply]
Ага, сорри за долги ответ
![image](https://github.com/VKCOM/VKUI/assets/67464545/dc43e442-41bf-403e-b9f3-4ae5b650a6f5)

**Reply by inomdzhon** [reply]
А `@types/react` есть у тебя в зависимостях проекта?

**Reply by scffs** [reply]
ой)) вот оно что))
